Assets/Scripts/Utility/AssetInjection/Components/CustomDoor.cs

```diff
@@ -4,89 +4,148 @@
-        /// <param name="dontCreateStaticDoors">If true, custom door components request to suppress classic doors.</param>
-        public static void InitDoors(GameObject building, StaticDoor[] staticDoors, int buildingKey, out bool dontCreateStaticDoors)
+        /// <param name="staticDoors">A list of static doors for the vanilla building for copying, if available.</param>
+        /// <param name="buildingKey">The key of the building that owns the exterior doors, -1 for interior doors.</param>
+        /// <param name="blockIndex">Block index for RMB doors.</param>
+        /// <param name="recordIndex">Record index of interior.</param>
+        /// <param name="buildingMatrix">Individual building matrix.</param>
+        /// <param name="disableClassicDoors">If true, at least one custom door component has requested suppression of all classic doors for this model.</param>
+        /// <returns>List of StaticDoor data for any custom doors that don't copy a DF model StaticDoor.</returns>
+        public static List<StaticDoor> InitDoors(GameObject building, StaticDoor[] staticDoors, int buildingKey, int blockIndex, int recordIndex, Matrix4x4 buildingMatrix, out bool disableClassicDoors)
         {
-            dontCreateStaticDoors = false;
-            // Set data to all doors in the building
+            // Iterate through all custom door components for this building
+            disableClassicDoors = false;
             CustomDoor[] allCustomDoors = building.GetComponentsInChildren<CustomDoor>();
+            List<StaticDoor> customStaticDoors = new List<StaticDoor>(allCustomDoors.Length);
             for (int i = 0; i < allCustomDoors.Length; i++)
             {
                 if (allCustomDoors[i].DisableClassicDoors == true)
-                    dontCreateStaticDoors = true;
+                    disableClassicDoors = true;
 
-                if ((allCustomDoors[i].StaticDoorCopied < 0) || (allCustomDoors[i].StaticDoorCopied > staticDoors.Length))
+                // If no static doors available (i.e. not replacing a DF model with door data) or not configured to copy one
+                if (allCustomDoors[i].StaticDoorCopied == -1 || staticDoors == null || staticDoors.Length == 0)
+                {
+                    // Calculate the door normal
+                    Vector3 doorCenter = allCustomDoors[i].DoorTrigger.center;
+                    Vector3 doorSize = allCustomDoors[i].DoorTrigger.size;
+                    Vector3 doorNormal = new Vector3();
+                    if (doorSize.x < doorSize.z && doorSize.x < doorSize.y)
+                        doorNormal.x = Mathf.Sign(doorCenter.x);
+                    else if (doorSize.z < doorSize.x && doorSize.z < doorSize.y)
+                        doorNormal.z = Mathf.Sign(doorCenter.z);
+                    else if (doorSize.y < doorSize.x && doorSize.y < doorSize.z)
+                        doorNormal.y = Mathf.Sign(doorCenter.y);
```

Comment:
@KABoissonneault This part calculating the door normal may not be how it should be done, but it does seem to work. This was my 4th revision of this code and I am open to better methods if you want to suggest anything. My 3d graphics knowledge is rudimentary at best.